jujugui/static/gui/src/app/store/env/base.js

```diff
@@ -275,10 +275,13 @@ YUI.add('juju-env-base', function(Y) {
 
       @method close
     */
-    close: function() {
+    close: function(cb) {
       if (this.ws) {
         this.beforeClose(this.ws.close.bind(this.ws));
       }
+      if (cb) {
+        cb();
```

Comment:
I was wrong, we definitely do want to because we need a websocket for the next connection so lets make all of this a little more explicit.